sopel/tools/memories.py

```diff
@@ -221,19 +224,19 @@ def _convert_keys(
         # return converted input data
         return ((self.make_identifier(k), v) for k, v in data)
 
-    def __getitem__(self, key: Optional[str]):
+    def __getitem__(self, key: str | None) -> Any:
         return super().__getitem__(self._make_key(key))
 
-    def __contains__(self, key):
+    def __contains__(self, key: Any) -> Any:
         return super().__contains__(self._make_key(key))
 
-    def __setitem__(self, key: Optional[str], value):
+    def __setitem__(self, key: str | None, value: Any) -> Any:
```

Comment:
Nit: that can be `None`

```suggestion
    def __setitem__(self, key: str | None, value: Any) -> None:
```

Reply:
Oh, nice catch.